theia-extensions/viewer-prototype/src/browser/trace-server-preference.ts

```diff
@@ -1,8 +1,8 @@
 import { PreferenceSchema, PreferenceProxy, PreferenceScope } from '@theia/core/lib/browser';
-import { TRACE_SERVER_DEFAULT_PORT } from '../common/trace-server-url-provider';
+import { TRACE_SERVER_DEFAULT_ARGS } from '../common/trace-server-url-provider';
 
-export const TRACE_PATH = 'trace-viewer.path';
-export const TRACE_PORT = 'trace-viewer.port';
+export const TRACE_PATH = 'trace Viewer.trace Server.path';
+export const TRACE_ARGS = 'trace Viewer.trace Server.arguments';
```

Comment:
The port preference should not be removed because it's needed in the URL of the REST API.

Reply:
Having the port specified in 2 fields is confusing and error prone. If Theia supported referencing preferences [1], perhaps in Arguments we could specify something like: `-vmargs -Dtraceserver.port=${config.trace Viewer.trace Viewer.port}`. But alas, looks like it's not implemented in Theia [2]. I guess we'll have to live with these 2 preferences for now, but at least we should modify the description of "Trace Server: Port" to clarify that it's purpose has changed and now it's used to specify what port to use by the client.

@paul-marechal, can you please confirm that  https://github.com/eclipse-theia/theia/issues/8805 isn't indeed implemented?

[1] https://code.visualstudio.com/docs/editor/variables-reference#_configuration-variables 
[2] https://github.com/eclipse-theia/theia/issues/8805

Reply:
Variable substitution in settings is a delicate matter, here's what VS Code does:
https://code.visualstudio.com/docs/editor/variables-reference#_is-variable-substitution-supported-in-user-and-workspace-settings

We might support as much as VS Code, but generically we can't use variables, let alone refer to other settings.

You can always manually substitute string patterns at use time in your logic though.

Reply:
yeah, I understand it seems confusing and error prone to specify the port at 2 different places. However, to make the way a server is launched independent of the actual server implementation, I think there is no other way. The front-end needs to be told which port to use and hence there is a separate preference.

Alternatively, we could standardize the the command-line which each trace server implementation needs to follow. It would be an API that needs to be supported (and probably versioned). For this solution the port is an obvious CLI parameter. The server workspace location is something else and might Trace Compass server specific.

Maybe we could inspired by how language servers are spawn, configured (port, cache for indexing etc).